test_utils/AssertAsync.js

```diff
@@ -40,6 +40,21 @@ class AssertAsync {
       );
     }
   }
+
+  static async reject(promise, message) {
+    try {
+      await promise;
+      throw new TypeError('Promise must reject');
```

Comment:
Is this really a type error? It is still a Promise, or not? 🤔

Reply:
Out of 6 possible JS error, TypeError seems correct choice with the logic that the promise type is wrong in a sense it must reject. 
Do you have other ideas?

Reply:
It could also be just `Error`. I don't mind if you leave it.

Reply:
Done